### PR TITLE
Update .ko.yaml based on the publish task 🌮

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This also removes the unused .ko.yaml.release file. We re-create a `.ko.yaml` during the release task.
Related to #2510 (but doesn't fix it)

/cc @afrittoli @dibyom @bobcatfish @ImJasonH 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

File: .ko.yaml [hunk 1]
```diff
@@ -2,3 +2,11 @@ baseImageOverrides:
   github.com/tektoncd/pipeline/cmd/creds-init: gcr.io/tekton-nightly/github.com/tektoncd/pipeline/build-base:latest
   github.com/tektoncd/pipeline/cmd/git-init: gcr.io/tekton-nightly/github.com/tektoncd/pipeline/build-base:latest
   github.com/tektoncd/pipeline/cmd/entrypoint: busybox  # image must have `cp` in $PATH
+baseBuildOverrides:
+  github.com/tektoncd/pipeline/cmd/controller:
+    env:
+    - name: CGO_ENABLED
```
Comment (ImJasonH) [inline .ko.yaml:8]: Why is this necessary? Can you add a comment or link to more info?

Comment (vdemeester) [reply .ko.yaml:8]: https://github.com/tektoncd/pipeline/commit/f85c53f047095e09d2017552877cb3fbac885b1e
This got added in that commit. I'm just keeping "in sync" those two for the time being :wink:

Comment (bobcatfish) [reply .ko.yaml:8]: Maybe a comment to explain makes sense? Looking at https://github.com/tektoncd/pipeline/commit/f85c53f047095e09d2017552877cb3fbac885b1e I don't quite understand why this was added in that case either

**Comment by vdemeester** [comment]
Updated :wink:

**Comment by chmouel** [comment]
/lgtm

sounds good to me,
